initramfs/init.sh

```diff
@@ -87,6 +87,47 @@ boot_root()
 	exec switch_root /mnt/root "${init}"
 }
 
+find_and_run_update()
+{
+	echo "Checking for updates ..."
+	for dev in ${UPDATE_DEVICES}; do
+		if [ ! -b "${dev}" ]; then
+			continue
+		fi
+
+		echo "Attempting to mount ${dev}."
+		if ! mount -t f2fs,ext4,vfat,auto -o exec,noatime "${dev}" "${UPDATE_MOUNT}"; then
+			continue
+		fi
+
+		if [ ! -x "${UPDATE_SCRIPT}" ]; then
+			umount "${dev}"
+			echo "No executable update '${UPDATE_SCRIPT}' found on ${dev}, trying next."
+			continue
+		fi
+
+		echo "Found update on ${dev}, executing update ${UPDATE_SCRIPT}."
+		if ! "${UPDATE_SCRIPT}"; then
+			umount "${dev}"
+			echo "Update failed!"
+			critical_error
+			break;
+		fi
+
+		echo "Update finished, cleaning up."
+		if ! chmod -x "${UPDATE_SCRIPT}" || [ -x "${UPDATE_SCRIPT}" ]; then
+			umount "${dev}"
+			echo "Please remove update medium and power off."
```

Comment:
Should there be a sleep? Because the shutdown quickly following this will probably also clear the screen.

Reply:
as this is a debug feature, no sleep is needed here. but once we add user images, we probably want to do something like 'splash_image() { cat img > fb0; sleep 5s } to ensure an image is always visible for atleast 5 seconds.

In the above specific case, it depends on the text we receive.